### ISSUE TITLE
.env: no such file or directory

### ISSUE THREAD

**Issue by chihao10711** [joho/godotenv#213]
When I use ```go run``` it works well. However, when using ```go build``` couldn't be found .env.
Please help me with this?

**Comment by henrikac** [comment]
I just tried

```go
// imports

func main() {
  err := godotenv.Load()
  if err != nil {
    log.Fatal("Error loading .env file")
  }
}
```

and then ran `go run main.go` and `go build main.go && ./main` without any issues.

Could you provide some more details?

**Comment by VentGrey** [comment]
Happening to me as well. Running the binary doesn't work. I'm using docker running a compiled binary and this is my faulty code:

```go
	logger := utils.NewLogger("main")

	err := godotenv.Load()

	if err != nil {
		logger.Log(
			utils.Error,
			"Could not load the required .env file",
			err,
		)
		os.Exit(1)
	}
```

The container console shows there is indeed a .env file:

```
root@9639525c6a1c:/app# ls -lah
total 15M
drwxr-xr-x  3 root root  105 Aug 20 03:02 .
dr-xr-xr-x 18 root root 4.0K Aug 19 22:23 ..
-rw-r--r--  1 root root  338 Aug 19 21:46 .env
-rw-r--r--  1 root root    1 Aug 20 03:02 .lastposition
-rw-r--r--  1 root root 1.1K Aug 14 23:12 config.json
drwxr-xr-x  2 root root   28 Aug 19 22:23 examples
-rwxr-xr-x  1 root root  15M Aug 19 20:10 hole
-rw-r--r--  1 root root 4.3K Aug 17 23:35 logfaker.pl
```

My logger reports:

```
[ERROR] - 2023-08-19 22:16:12 : Could not load the required .env file open .env: no such file or directory - At package: main
```

Any ideas on what might have happened?

**Comment by arieansyah** [comment]
> When I use `go run` it works well. However, when using `go build` couldn't be found .env. Please help me with this?

try this : 
`godotenv.Load("full_path")`

`go clean`
`go build`

**Comment by coolaj86** [comment]
It looks like this is as stale issue that can be closed?

cc/ @chihao10711

**Comment by beiriannydd** [comment]
so my expectation prior to reading the code would be that it would _look_ for .env files in the list you gave it, loading the ones that exist, so that you could give it a list and it would load what it found.  The reality is that it stops at the first missing file, due to a file not found error being treated as a hard error.  This is a design decision, but is it what the user wants?  I ended up writing a
function to filter the list to only existing files before passing to load, but of course this won't work with autoload.
I look in the current directory for .env.${ENVIRONMENT}, .env, then $(dirname ${GOMOD}) for the same.

Basically the Load() function here works like the ruby Load!() which I wouldn't expect given the project goal.

**Comment by gwthm-in** [comment]
@beiriannydd Even I have similar requirements to try loading all files found and return an error at the end if any errors occur along with the ability to look at the root git directory, also at the go mod directory level. 

Probably if you have similar requirements, you can utilise [this](https://github.com/gwthm-in/dotenv)
